test/async-test.js

```diff
@@ -0,0 +1,204 @@
+var assert, fs, path, protagonist;
+
+fs = require('fs');
+path = require('path');
+assert = require('chai').assert;
+protagonist = require('./protagonist');
+
+const data = fs.readFileSync(path.join(__dirname, './fixtures/sample-api.apib'), 'utf8');
+const sync_parsed = protagonist.parseSync(data)
+const sync_parsed_sm = protagonist.parseSync(data, { exportSourcemap : true })
+
+describe('Async parse', () => {
+
+  it('will parse with callback', function(done) {
+    protagonist.parse(data, function (err, res) {
+      assert.isNull(err)
+      assert.deepEqual(res, sync_parsed);
+      done()
+    })
+  })
+
+  it('will parse with promise', function() {
+    protagonist.parse(data)
+      .then(res => {
+        assert.deepEqual(res, sync_parsed);
+      })
+      .catch(err => {
+        assert.fail()
+      })
+
+  })
+
+  it('will parse with await/async', async function() {
```

Comment:
Since we're still supporting Node 4/6 and this (async function) isn't valid syntax, we're going to have to disable testing async/await on those versions of Node. I'd suggest we do something like the following, although there are perhaps other options which can be explored:

1. Move the async/await tests out to separate file (test-async-await.js or similiar).
2. Change package.json so that the test line (https://github.com/apiaryio/protagonist/blob/bcfba75c652666143273944347e5871b0a7b7ef0/package.json#L25) adds the argument `--exclude test/test-async-await.js`.
3. On Circle CI config, enable this specific test, add extra run step after https://github.com/apiaryio/protagonist/blob/bcfba75c652666143273944347e5871b0a7b7ef0/.circleci/config.yml#L27 to do `- run: ./node_modules/.bin/mocha test/test-async-await.js` so that these tests are enable on Circle CI environment on the newer node versions.

Reply:
Yes. I have seen. Separating async/await tests and prepate condtional test running for node > 6